From e451b6a8baec970183ffa6b1cc12d36e3d25fcd5 Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Wed, 26 Feb 2014 12:03:36 +0100
Subject: [PATCH 5/7] Allow overriding flash offsets.

Modify zynq-xommon.h to allow a board include to specify which
addresses to use in flash (because of sector alignment).

Name bitstream "fpga.bin". Remove unused environment vars.

Adapt zynq_miami.h accordingly, so the ENV need not be repeated.
---
 include/configs/zynq-common.h |   24 +++++++++++--
 include/configs/zynq_miami.h  |   76 ++++++-----------------------------------
 2 files changed, 31 insertions(+), 69 deletions(-)

diff --git a/include/configs/zynq-common.h b/include/configs/zynq-common.h
index 1477d36..ec10f41 100644
--- a/include/configs/zynq-common.h
+++ b/include/configs/zynq-common.h
@@ -10,6 +10,10 @@
 #ifndef __CONFIG_ZYNQ_COMMON_H
 #define __CONFIG_ZYNQ_COMMON_H
 
+/* C stuff to convert macro contents to quoted string */
+#define QUOTE(name) #name
+#define STR(macro) QUOTE(macro)
+
 /* High Level configuration Options */
 #define CONFIG_ARMV7
 #define CONFIG_ZYNQ
@@ -231,6 +235,12 @@
 #ifndef CONFIG_EXTRA_ENV_BOOTSCRIPT
 # define CONFIG_EXTRA_ENV_BOOTSCRIPT "autorun.scr"
 #endif
+#ifndef CONFIG_SYS_SPI_DEVTREE_OFFS
+# define CONFIG_SYS_SPI_DEVTREE_OFFS 0x70000
+#endif
+#ifndef CONFIG_SYS_SPI_UIMAGE_OFFS
+# define CONFIG_SYS_SPI_UIMAGE_OFFS	0x80000
+#endif
 
 /* Default environment */
 #define CONFIG_EXTRA_ENV_SETTINGS	\
@@ -258,8 +268,8 @@
 		"fpga load 0 ${loadbit_addr} ${filesize}\0" \
 	"qspiboot=echo Booting from QSPI flash... && " \
 		"sf probe 0 0 0 && " \
-		"sf read 0x2A00000 0x70000 ${devicetree_size} && " \
-		"sf read ${kernel_addr} 0x80000 ${kernel_size} && " \
+		"sf read 0x2A00000 " STR(CONFIG_SYS_SPI_DEVTREE_OFFS) " ${devicetree_size} && " \
+		"sf read ${kernel_addr} " STR(CONFIG_SYS_SPI_UIMAGE_OFFS) " ${kernel_size} && " \
 		"bootm ${kernel_addr} - 0x2A00000\0" \
 	"uenvboot=" \
 		"if run loadbootenv; then " \
@@ -447,7 +457,9 @@
 #define CONFIG_SPL_SPI_LOAD
 #define CONFIG_SPL_SPI_FLASH_SUPPORT
 #define CONFIG_SPL_SPI_BUS	0
+#ifndef CONFIG_SYS_SPI_U_BOOT_OFFS
 #define CONFIG_SYS_SPI_U_BOOT_OFFS	0x10000
+#endif
 #define CONFIG_SPL_SPI_CS	0
 #endif
 
@@ -466,7 +478,13 @@
 #define CONFIG_SPL_TEXT_BASE	0x0
 
 /* 3 * 64kB blocks of OCM - one is on the top because of bootrom */
-#define CONFIG_SPL_MAX_FOOTPRINT	0x10000
+#ifndef CONFIG_SPL_MAX_FOOTPRINT
+# if (CONFIG_SYS_SPI_U_BOOT_OFFS < 0x30000)
+#  define CONFIG_SPL_MAX_FOOTPRINT	CONFIG_SYS_SPI_U_BOOT_OFFS
+# else
+#  define CONFIG_SPL_MAX_FOOTPRINT	0x30000
+# endif
+#endif
 
 /* The highest 64k OCM address */
 #define OCM_HIGH_ADDR	0xffff0000
diff --git a/include/configs/zynq_miami.h b/include/configs/zynq_miami.h
index 2818dd3..2908565 100644
--- a/include/configs/zynq_miami.h
+++ b/include/configs/zynq_miami.h
@@ -13,82 +13,27 @@
 #define CONFIG_SYS_SDRAM_SIZE		(512 * 1024 * 1024)
 #define CONFIG_ZYNQ_PS_CLK_FREQ		32000000UL
 #define CONFIG_ZYNQ_SERIAL_UART1
-/* Ethernet needs external clock, not really supported here yet but
- * it won't compile without it. */
+/* Ethernet (requires logic to actually work). */
 #define CONFIG_ZYNQ_GEM0
-#define CONFIG_ZYNQ_GEM_PHY_ADDR0 0
-#undef CONFIG_ZYNQ_GEM1
-#undef CONFIG_SYS_ENET
+#define CONFIG_ZYNQ_GEM_PHY_ADDR0 3
 
+/* No NOR flash */
 #define CONFIG_SYS_NO_FLASH
-
+/* SD and QSPI support */
 #define CONFIG_ZYNQ_SDHCI0
 #define CONFIG_ZYNQ_QSPI
 #define CONFIG_ZYNQ_USB
 
-#define CONFIG_ZYNQ_BOOT_FREEBSD
 #define CONFIG_DEFAULT_DEVICE_TREE	zynq-miami
 
-
-#include <configs/zynq-common.h>
-
-#ifndef CONFIG_EXTRA_ENV_RSA_BOOT
-# define CONFIG_EXTRA_ENV_RSA_BOOT
-#endif
-
 /* Different SPI flash partitions because of the 256k erase size */
 #define CONFIG_SYS_SPI_U_BOOT_OFFS	0x40000
-#define CONFIG_SPL_MAX_FOOTPRINT	0x30000
-#define CONFIG_EXTRA_ENV_SETTINGS	\
-	"ethaddr=00:0a:35:00:01:22\0"	\
-	"kernel_image=uImage\0"	\
-	"ramdisk_image=uramdisk.image.gz\0"	\
-	"devicetree_image=devicetree.dtb\0"	\
-	"bitstream_image=system.bit.bin\0"	\
-	"bootscript=" CONFIG_EXTRA_ENV_BOOTSCRIPT "\0" \
-	"loadbit_addr=0x100000\0"	\
-	"loadbootenv_addr=0x2000000\0" \
-	"kernel_size=0x380000\0"	\
-	"kernel_addr=0x79C0\0"	\
-	"devicetree_size=0x10000\0"	\
-	"boot_size=0xF00000\0"	\
-	"fdt_high=0x20000000\0"	\
-	"initrd_high=0x20000000\0"	\
-	"bootenv=uEnv.txt\0" \
-	"loadbootenv=fatload mmc 0 ${loadbootenv_addr} ${bootenv}\0" \
-	"importbootenv=echo Importing environment from SD ...; " \
-		"env import -t ${loadbootenv_addr} $filesize\0" \
-	"mmc_loadbit_fat=echo Loading bitstream from SD/MMC/eMMC to RAM.. && " \
-		"mmcinfo && " \
-		"fatload mmc 0 ${loadbit_addr} ${bitstream_image} && " \
-		"fpga load 0 ${loadbit_addr} ${filesize}\0" \
-	"qspiboot=echo Booting from QSPI flash... && " \
-		"sf probe 0 0 0 && " \
-		"sf read 0x2A00000 0x140000 ${devicetree_size} && " \
-		"sf read ${kernel_addr} 0x180000 ${kernel_size} && " \
-		"bootm ${kernel_addr} - 0x2A00000\0" \
-	"uenvboot=" \
-		"if run loadbootenv; then " \
-			"echo Loaded environment from ${bootenv}; " \
-			"run importbootenv; " \
-		"fi; " \
-		"if test -n $uenvcmd; then " \
-			"echo Running uenvcmd ...; " \
-			"run uenvcmd; " \
-		"fi\0" \
-	"sdboot=if mmcinfo; then " \
-			"run uenvboot; " \
-			"echo Copying Linux from SD to RAM... && " \
-			"fatload mmc 0 ${kernel_addr} ${kernel_image} && " \
-			"fatload mmc 0 0x2A00000 ${devicetree_image} && " \
-			"bootm ${kernel_addr} - 0x2A00000\0" \
-		"fi\0" \
-	"jtagboot=echo TFTPing Linux to RAM... && " \
-		"tftp ${kernel_addr} ${kernel_image} && " \
-		"tftp 0x2A00000 ${devicetree_image} && " \
-		"tftp 0x2000000 ${ramdisk_image} && " \
-		"bootm ${kernel_addr} 0x2000000 0x2A00000\0" \
-	CONFIG_EXTRA_ENV_RSA_BOOT
+#define CONFIG_ENV_OFFSET			0x100000
+#define CONFIG_ENV_SECT_SIZE		0x40000
+#define CONFIG_SYS_SPI_DEVTREE_OFFS 0x140000
+#define CONFIG_SYS_SPI_UIMAGE_OFFS	0x180000
+
+#include "zynq-common.h"
 
 /* Ethernet PHY is different */
 #undef CONFIG_PHY_MARVELL
@@ -97,6 +42,5 @@
 
 /* Extras */
 #define CONFIG_CMD_MEMTEST
-#define CONFIG_CMD_EXT4
 
 #endif /* __CONFIG_ZYNQ_MIAMI_H */
-- 
1.7.9.5

